File: code_combat/8_Cloudrip_Mountain/441-Mixed_Unit_Tactics/mixed_unit_tactics.py
summonTypes = False

def summonTroops():
    typeTo = "soldier"
    if summonTypes:
        typeTo = "archer"
    if hero.gold >= hero.costOf(typeTo):
        hero.summon(typeTo)
        summonTypes = not summonTypes


def commandAll():
    friends = hero.findFriends()
    for friend in friends:
        if friend.type == "archer" or friend.type == "soldier":
            enemy = friend.findNearestEnemy()
            if enemy:
                hero.command(friend, "attack", enemy)


def collectGold():
    coin = hero.findNearestItem()
    if coin:
        hero.move(coin.pos)


while True:
    collectGold()
    summonTroops()
    commandAll()

File: code_combat/8_Cloudrip_Mountain/441-Mixed_Unit_Tactics/mixedUnitTactics.js
// Practice using modulo to loop over an array

// Choose the mix and order of units you want to summon by populating this array:
var summonTypes = false;

function summonTroops() {
    // Use % to wrap around the summonTypes array based on hero.built.length
    //var type = summonTypes[???];
    var type = "soldier";
    if (summonTypes)
        type = "archer";
    if (hero.gold >= hero.costOf(type)) {
        hero.summon(type);
        summonTypes = !summonTypes;
    }
}


function commandAll() {
    var friends = hero.findFriends();
    for (var i = 0; i < friends.length; i++) {
        var friend = friends[i];
        if (friend.type == "archer" || friend.type == "soldier") {
            var enemy = friend.findNearestEnemy();
            if (enemy)
                hero.command(friend, "attack", enemy);
        }
    }
}


function collectGold() {
    var coin = hero.findNearestItem();
    if (coin)
        hero.move(coin.pos);
}


while (true) {
    collectGold();
    summonTroops();
    commandAll();
}
